### PR TITLE
CC fields missing on checkout page for logged in customer with no available saved cards for selected currency

### PR DESCRIPTION
**Pull request by ChantalDelachevrotiere**
Added logic for when no contracts are available for selected currency to display CC fields

### CODE REVIEW

**Review by grandel-withreach — APPROVED** [review]

File: view/frontend/web/js/view/payment/method-renderer/cc.js [hunk 1]
```diff
@@ -37,7 +37,9 @@ define([
                         contractOptions.push(item);
                     }
                 });
-                contractOptions.push({contractId:0,label:$t('Use Different Card')});
+                if (contractOptions.length){
```
Comment (grandel-withreach) [inline view/frontend/web/js/view/payment/method-renderer/cc.js:40]: SHould this be `.length > 0` or osmething like that?

**Review by sravantiv — APPROVED** [review]

File: view/frontend/web/js/view/payment/method-renderer/cc.js [hunk 2]
```diff
@@ -37,7 +37,9 @@ define([
...
+                    contractOptions.push({contractId:0,label:$t('Use Different Card')});
```
Comment (grandel-withreach) [inline view/frontend/web/js/view/payment/method-renderer/cc.js]: Could we also put these in separate lines for readability?

ex.

```
{
     contractId : 0,
     label : $t('Use Different Card')
}
```

**Review by grandel-withreach — CHANGES_REQUESTED** [review]
Just made additional comments that I missed the first time

File: view/frontend/web/js/view/payment/method-renderer/cc.js (hunk 1 above)
Comment (ChantalDelachevrotiere) [reply view/frontend/web/js/view/payment/method-renderer/cc.js:40]: Similar syntax is used above on line 32 `window.checkoutConfig.payment.reach_cc.open_contracts.length`

File: view/frontend/web/js/view/payment/method-renderer/cc.js (hunk 2 above)
Comment (ChantalDelachevrotiere) [reply view/frontend/web/js/view/payment/method-renderer/cc.js]:
```
                if (contractOptions.length){
                    contractOptions.push(
                        {
                            contractId:0,
                            label:$t('Use Different Card')
                        }
                        );
                }
```
? :)

Comment (ChantalDelachevrotiere) [reply view/frontend/web/js/view/payment/method-renderer/cc.js]: phpstorm wants it to look like this

Comment (grandel-withreach) [reply view/frontend/web/js/view/payment/method-renderer/cc.js]: Really? If wont accept the following?

```
if (contractOptions.length){
     contractOptions.push({
            contractId:0,
            label:$t('Use Different Card')
     });
 }
```

**Review by grandel-withreach — APPROVED** [review]

**Pull request merged**